viewer/packages/cad-geometry-loaders/src/sector/SectorLoader.ts

```diff
@@ -88,8 +88,10 @@ export class SectorLoader {
       const filteredSectors = await this.filterSectors(sectorCullerInput, batch, sectorCuller, progressHelper);
       const consumedPromises = this.startLoadingBatch(filteredSectors, progressHelper, cadModels);
       for await (const consumed of PromiseUtils.raceUntilAllCompleted(consumedPromises)) {
-        this._modelStateHandler.updateState(consumed);
-        yield consumed;
+        if (consumed.result) {
```

Comment:
Won't this fail horribly if the result valid, but is falsy? E.g. 0, false etc etc

Reply:
`consumed` is supposed to be of the type `PromiseResult<ConsumedSector>`, and `result` will have type `ConsumedSector`. If it is falsy, it does not have that type, and I think it's fair to treat it as an error in that case.

Reply:
Ok, I still think being explicit about this is best - so I still prefer this:
```suggestion
        if (consumed.result !== undefined) {
```
up to you